src/PHPRedisSequenceResolver.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the godruoyi/php-snowflake.
+ *
+ * (c) Godruoyi <g@godruoyi.com>
+ *
+ * This source file is subject to the MIT license that is bundled.
+ */
+
+namespace Godruoyi\Snowflake;
+
+use Predis\Client as PredisClient;
+
+class PHPRedisSequenceResolver implements SequenceResolver
+{
+    /**
+     * The cache prefix.
+     */
+    protected string $prefix = '';
+
+    /**
+     * The default redis lua script
+     */
+    protected static string $script = <<<'LUA'
+if redis.call('set', KEYS[1], ARGV[1], "EX", ARGV[2], "NX") then
+    return 0
+else
+    return redis.call('incr', KEYS[1])
+end
+LUA;
```

Comment:
This looks ugly aligned along the hard-left edge, but could (and should) be indented two tab stops with [flexible heredoc syntax](https://wiki.php.net/rfc/flexible_heredoc_nowdoc_syntaxes) (supported since PHP 7.3). Applies throughout this PR.

```suggestion
        if redis.call('set', KEYS[1], ARGV[1], "EX", ARGV[2], "NX") then
            return 0
        else
            return redis.call('incr', KEYS[1])
        end
        LUA;
```